### PR TITLE
mock endpoint values

### PR DESCRIPTION
**Pull request by Manoj-nathwani**
WIP API changes to allow front-end testing locally

### CODE REVIEW

File: navigator_api/main.py [hunk 1]
```diff
@@ -27,23 +31,97 @@ def user_details():
...
+    }
+)
+def dataset_state_update(post_body):
+    logging.info(post_body)
+    return jsonify({'success': True})
+    # dataset_state = db.get_dataset_state(post_body['dataset_id'])
+    # if dataset_state['last_updated'] != post_body['last_updated']:
+    #     return jsonify({
+    #         'message': 'Another user has already updated this dataset'
+    #     }), 500
```
Comment (Manoj-nathwani) [inline navigator_api/main.py:128]: @tomeksabala I hope it's clear what i'm doing here.
Race-conditions between users clicking the `Next Step` button is something we need to solve for

File: navigator_api/main.py [hunk 2]
```diff
@@ -27,23 +31,97 @@ def user_details():
...
+        ]
+    )
+
+
+@main_blueprint.route('/dataset_state', methods=['GET'])
+# @login_required
+@use_args(
+    {'dataset_id': fields.Int(required=True)},
+    location='query'
+)
```
Comment (tomeksabala) [inline navigator_api/main.py]: that should be simple `@main_blueprint.route('/dataset_state/<dataset_id>', methods=['GET'])`

File: navigator_api/main.py [hunk 3]
```diff
@@ -27,23 +31,97 @@ def user_details():
...
+            {
+                "id": "3",
+                "name": "Antarctica Inputs UNAIDS Estimates 2021"
+            }
+        ]
+    )
+
+
+@main_blueprint.route('/dataset_state', methods=['GET'])
+# @login_required
```
Comment (tomeksabala) [inline navigator_api/main.py]: why you're ignoring the auth?

File: navigator_api/main.py [hunk 4]
```diff
@@ -27,23 +31,97 @@ def user_details():
...
-                    "id": "2",
-                    "name": "Malawi Inputs UNAIDS Estimates 2021"
-                },
-                {
-                    "id": "3",
-                    "name": "Antarctica Inputs UNAIDS Estimates 2021"
-                }
-            ]
+            'task_number': task_number,
+            'milestone': possible_milestones[randint(0, len(possible_milestones)-1)],
```
Comment (tomeksabala) [inline navigator_api/main.py:101]: I'd avoid using random number is mocked/demo data.

File: navigator_api/main.py [hunk 5]
```diff
@@ -27,23 +31,97 @@ def user_details():
...
+    ]
+    list_items = ('').join([
+        f'<li>{get_sentence(count=(1, 3))}</li>'
+        for x in range(randint(1, 7))
+    ])
+    html = ''.join([
+        f'<h3>{task_name}</h3>',
+        f'<p>{get_paragraph(count=(3, 5))}</p>',
+        f'<ol>{list_items}</ol>'
+    ])
```
Comment (tomeksabala) [inline navigator_api/main.py]: That could be a jinja2 template. I think it'll be useful in the future so it's wort refactoring at this stage.

File: navigator_api/main.py (hunk 1 above)
Comment (tomeksabala) [reply navigator_api/main.py:128]: yeah, I understand. I'll need to think for a bit how to handle it.

**Review by tomeksabala — APPROVED** [review]

File: navigator_api/main.py (hunk 5 above)
Comment (Manoj-nathwani) [reply navigator_api/main.py]: I'm mocking the html which will eventually be fetched from the database. So this is good enough for now

File: navigator_api/main.py (hunk 3 above)
Comment (Manoj-nathwani) [reply navigator_api/main.py]: This is a WIP, but it allowed me to test the app locally (outside of minkube)
Today I will be getting this app working inside minikube and re-enabling auth 🤓

**Pull request merged**